src/main/scala/edu/ie3/mobsim/MobilitySimulator.scala

```diff
@@ -302,9 +307,12 @@ final class MobilitySimulator(
         cs -> movement
       } match {
         case result @ Some(_) =>
-          ev.setChargingAtSimona(false)
-          ev.setChosenChargingStation(None)
-          result
+          val updatedEv: ElectricVehicle =
+            ev.removeChargingAtSimona().setChosenChargingStation(None)
+
+          val cs: UUID = result.value._1
+
+          Some(cs, MobilitySimulator.Movement(cs, updatedEv))
```

Comment:
This method needs some more refactoring imho. The movement created in line 303 just goes to waste and can be removed. In turn, all of these ev updates can go into the mapping part from line 302-306. Hit me up if this needs discussion :)